src/CategoryLauncher.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import "./CategoryLauncher.css";
+import { isValidSlug } from "./Loader";
+type CLProps = {
+	setTableSlug: (slug: string, validity: boolean) => Promise<void>;
+	setTableGoal: (goal: string) => void;
+	goalRaceMap: Map<string, Set<number>>;
+};
+
+export default function CategoryLauncher(props: CLProps) {
+	const { setTableSlug, setTableGoal, goalRaceMap } = props;
+
+	const updateSlug = (newSlug: string) => {
+		setSlug(newSlug);
+		isValidSlug(newSlug).then((valid) => {
+			setValidSlug(valid);
+		});
```

Comment:
Prefer `await` over `.then`.

The reliance on component level state makes it easy to run into race conditions. A fun race condition here is to enter a valid slug, then add a character and immediately submit before the validity could be checked, at which point you get a full screen error message about a NetworkError.

This might be an argument in favor of pulling the validation logic up and making this component fully controlled, though it's not the only way to fix this of course.